com.unity.ml-agents/Runtime/InferenceBrain/ModelRunner.cs

```diff
@@ -11,16 +11,13 @@ public struct AgentInfoSensorsPair
         public AgentInfo agentInfo;
         public List<ISensor> sensors;
     }
-    public struct AgentIdActionPair
-    {
-        public int agentId;
-        public Action<AgentAction> action;
-    }
 
     public class ModelRunner
     {
         List<AgentInfoSensorsPair> m_Infos = new List<AgentInfoSensorsPair>();
-        List<AgentIdActionPair> m_ActionFuncs = new List<AgentIdActionPair>();
+        Dictionary<int, float[]> m_LastActionsReceived = new Dictionary<int, float[]>();
+        List<int> m_ActionId = new List<int>();
```

Comment:
Same comment about m_AgentsRequestingDecisions (and in a few argument names too). Feel free to call something different, but I don't think ActionId is meaningful